### PR TITLE
Added module to longhaul tests to cleanup orphaned hub device resources

### PR DESCRIPTION
**Pull request by ericwol-msft**
# Checklist
- [x] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/master/.github/CONTRIBUTING.md).
- [x] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [x] This pull-request is submitted against the `master` branch. 
  - [x] I have merged the latest `master` branch prior to submission and re-merged as needed after I took any feedback.
  - [x] I have squashed my changes into one with a clear description of the change.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

File: iothub_client/tests/longhaul_resource_cleanup/longhaul_resource_cleanup.c [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+#include "iothub_account.h"
+#include "iothubtest.h"
+#include "parson.h"
+#include "../common_longhaul/iothub_client_common_longhaul.h"
+
+#ifdef SET_TRUSTED_CERT_IN_SAMPLES
+#include "certs.h"
+#endif // SET_TRUSTED_CERT_IN_SAMPLES
+
+const int five_days = (60 * 60 * 24) * 5;
```
Comment (ewertons) [inline iothub_client/tests/longhaul_resource_cleanup/longhaul_resource_cleanup.c:26]: I have no good comments, you got all covered. So here it goes... please use all upper case for constants.

```suggestion
const int FIVE_DAYS = (60 * 60 * 24) * 5;
```

File: iothub_client/tests/longhaul_resource_cleanup/longhaul_resource_cleanup.c [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+const int year_2020 = 1577865600;
```
Comment (ewertons) [inline iothub_client/tests/longhaul_resource_cleanup/longhaul_resource_cleanup.c:27]:
```suggestion
const int YEAR_2020 = 1577865600;
```

File: iothub_client/tests/longhaul_resource_cleanup/longhaul_resource_cleanup.c [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+{
+    const ENUM_CONTEXT* context = (const ENUM_CONTEXT*) action_context;
+    const IOTHUB_MODULE* device = (const IOTHUB_MODULE*) item;
+    if (strncmp(device->deviceId, SAS_DEVICE_PREFIX_FMT, sizeof(SAS_DEVICE_PREFIX_FMT) - 1) == 0 || 
+        strncmp(device->deviceId, X509_DEVICE_PREFIX_FMT, sizeof(X509_DEVICE_PREFIX_FMT) - 1) == 0)
+    {
+        char* twin = IoTHubDeviceTwin_GetTwin(context->deviceTwinHandle, device->deviceId);
+        if (twin)
+        {
+            JSON_Object* root_object = NULL;
```
Comment (ewertons) [inline iothub_client/tests/longhaul_resource_cleanup/longhaul_resource_cleanup.c:46]: Are any of these variables dynamically allocated? If so, I don't see any frees afterwards.

**Review by ewertons — APPROVED** [review]

File: iothub_client/tests/longhaul_resource_cleanup/longhaul_resource_cleanup.c (hunk 3 above)
Comment (ericwol-msft) [reply iothub_client/tests/longhaul_resource_cleanup/longhaul_resource_cleanup.c:46]: it's freed by json_value_free(root_value);

Comment (ericwol-msft) [reply iothub_client/tests/longhaul_resource_cleanup/longhaul_resource_cleanup.c:46]: see https://github.com/kgabis/parson

File: iothub_client/tests/longhaul_resource_cleanup/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    ${PROJECT_NAME}.c
+)
+
+set(project_h_files
+    ${PROJECT_NAME}.h
+)
+
+build_c_test_longhaul_test(${PROJECT_NAME} ${project_c_files} ${project_h_files})
+
+target_link_libraries(${PROJECT_NAME})
```
Comment (danewalton) [inline iothub_client/tests/longhaul_resource_cleanup/CMakeLists.txt:20]: I don't believe this is doing anything

File: iothub_client/tests/longhaul_resource_cleanup/longhaul_resource_cleanup.c [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+    SINGLYLINKEDLIST_HANDLE deviceList = NULL;
+
+    if ((iotHubLonghaulRsrcsHandle = longhaul_tests_init()) == NULL)
+    {
+        LogError("Test failed");
+        result = MU_FAILURE;
+    }
+    else if ((account_info = longhaul_get_account_info(iotHubLonghaulRsrcsHandle)) == NULL)
+    {
+        LogError("Failed retrieving accoung info");
```
Comment (danewalton) [inline iothub_client/tests/longhaul_resource_cleanup/longhaul_resource_cleanup.c:121]:
```suggestion
        LogError("Failed retrieving account info");
```

File: iothub_client/tests/longhaul_resource_cleanup/longhaul_resource_cleanup.h [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* LONGHAUL_RESOURCE_CLEANUP_H */
```
Comment (danewalton) [inline iothub_client/tests/longhaul_resource_cleanup/longhaul_resource_cleanup.h:16]: curious the point of this file?

**Review by danewalton — APPROVED** [review]

File: iothub_client/tests/longhaul_resource_cleanup/CMakeLists.txt (hunk 4 above)
Comment (ericwol-msft) [reply iothub_client/tests/longhaul_resource_cleanup/CMakeLists.txt:20]: me too

File: iothub_client/tests/longhaul_resource_cleanup/longhaul_resource_cleanup.h (hunk 6 above)
Comment (ericwol-msft) [reply iothub_client/tests/longhaul_resource_cleanup/longhaul_resource_cleanup.h:16]: linux-options fails without it. Same goes with the other longhaul tests https://azure-iot-sdks.visualstudio.com/azure-iot-sdks/_build/results?buildId=79981&view=logs&j=9157c21d-e235-5a95-355e-08c3ffae2e82

**Pull request merged**